sdk/monitor/Azure.Monitor.Query/src/QueryTimeSpan.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+
+namespace Azure.Monitor.Query
+{
+    /// <summary>
+    /// Represents a span of time over which the query would be executed.
+    /// </summary>
+    public readonly struct QueryTimeSpan : IEquatable<QueryTimeSpan>
```

Comment:
I'm not happy with the name but also don't have good ideas. I tried Interval but the Query API has another thing called interval (sampling interval)

Reply:
Have you considered `TimeSpanRange`?

Reply:
`DateTimeRange` is nice... I like that.